### PR TITLE
Users & groups db prep

### PR DESCRIPTION
**Pull request by dannyroberts**
@snopoke 

Split views out to make it easy to move users and groups to their own db

cc: @czue @nickpell

### CODE REVIEW

**Comment by snopoke** [comment]
:white_check_mark: probably good to wait till after summit to merge?

**Comment by dannyroberts** [comment]
Yeah probably

File: corehq/couchapps/all_docs/views/by_doc_type/map.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
+function (doc) {
```
Comment (dimagimon) [inline corehq/couchapps/all_docs/views/by_doc_type/map.js]: Missing name in function declaration.

**Comment by dannyroberts** [comment]
self-merging with simon's sign off
heads up @dimagi/team-commcare-hq, relatively big preindex (doesn't affect the highest volume doc tpyes, but affects bunch of views). preindexing prod now

File: corehq/apps/locations/dbaccessors.py [hunk 2]
```diff
@@ -2,7 +2,7 @@ def _cc_users_by_location(domain, location_id, include_docs=True, wrap=True):
     from corehq.apps.users.models import CommCareUser
     view = CommCareUser.view if wrap else CommCareUser.get_db().view
     return view(
-        'locations/users_by_location_id',
+        'users_extra/users_by_location_id',
```
Comment (esoergel) [inline corehq/apps/locations/dbaccessors.py]: :+1: It always felt weird that a view on user docs was in the locations module.

**Pull request merged**